### PR TITLE
Make examples/ scripts more script-like with flags and stuff

### PR DESCRIPTION
**Pull request by adamrothman**
They can also be run with `--help` to see what's available.

### CODE REVIEW

File: examples/backtest.py [hunk 1]
```diff
@@ -1,24 +1,62 @@
...
-print(summary)
+    fund = Fund(strategies[args.strategy], BacktestMarketAdapter, config)
+
+    summary = evaluate(
+        fund,
+        '2017-01-01',
+        '2017-06-29',
+        duration_days=30,
+        window_distance_days=14,
+    )
```
Comment (adamrothman) [inline examples/backtest.py]: The params here could easily be set with other CLI flags.

File: examples/backtest.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def main(args):
     )
     parser.add_argument(
         '-s', '--strategy',
-        default='buffed_coin',
+        default='buffed-coin',
```
Comment (elsehow) [inline examples/backtest.py]: What about in the future, when the user will be writing their own strategies/using this as a library? #11

File: examples/live_trading.py [hunk 3]
```diff
@@ -1,14 +1,52 @@
...
 from moneybot.fund import Fund
 from moneybot.markets.adapters.live import LiveMarketAdapter
 from moneybot.strategies.buffed_coin import BuffedCoinStrategy
+from moneybot.strategies.buy_hold import BuyHoldStrategy
+from moneybot.strategies.peak_rider import PeakRiderStrategy
 
 
-with open('config.json') as cfg_file:
-    config = json.load(cfg_file)
+strategies = {
```
Comment (elsehow) [inline examples/live_trading.py]: This makes sense for now, but see other comment about extensibility

**Comment by adamrothman** [comment]
Given that `backtest.py` and `live_trading.py` are in `examples/`, my goal was just to make it easy to demo the other strategies without having to modify the scripts. That said, I think I agree with you that strategies should be provided by the user. I guess in that case they would have to write their strategy implementation in the same file as their main script but that doesn't seem like a bad thing.

Agree that it makes sense to provide only the `Strategy` ABC as part of the library itself. I can move the subclasses into `examples/` but it would be great if you could update the docstrings (including the base one with instructions about how to subclass) so that people (me, lol) can better understand how they work.

In light of this change, how should the backtest and live trade examples work? `examples/` is outside the module structure so we can't import stuff from it without some hackery (doable though).

**Comment by adamrothman** [comment]
Having said all that though I feel like there is precedent for including example code as part of your library so that people can import it/start with it as a base. `¯\_(ツ)_/¯`

**Comment by elsehow** [comment]
I have been thinking about why the present situation confuses me. Here's what I *think* I'm reacting to:

When we pass in the 'strategy' flag, and look up strategies in that dict, it makes it seem that either

- We are enumerating all the strategies the library can do
- We are enumerating all the strategies the example can do

Right now, neither of these really apply.

We aren't enumerating all the library's strategies, because the library doesn't "have" strategies - it just provides a framework for writing them.

We also aren't enumerating all the strategies the example can do, because the example doesn't "have" strategies - it imports them from the library!

I can think of two solutions:

1. Put the strategies in the lib some way that emphasizes their "example" nature (like strategies.examples.buffed_coin or examples.strategies.buffed_coin or whatever).

2. Put the strategies in examples/

I am not sure which I prefer. OTOH, (1) allows us to ship strategies with the library, which can be nice, as you mention. OTOH, (2) gives a clear example for how you, the prospective strategy-designer, can import and subclass the library for your own purposes.

Thoughts?

@harperj @JakeHartnell

**Comment by adamrothman** [comment]
Having given it a little more thought, I think I prefer solution 1. It makes clear to anyone importing the strategies that they are examples while making them easy to work with as a starting point.

I get what you're saying about selecting a strategy with a flag but – I think it's OK as long as `backtest.py` and `live_trading.py` live in `examples/`. The scripts themselves are examples, which draw on the library's example strategies, to give you an example of how to use the library in an end-to-end way.

**Comment by elsehow** [comment]
Agreed! If the strategies are in `strategies.examples`, it makes sense to have dict in the example scripts, to make it more obvious what other strategies to explore.

**Comment by harperj** [comment]
The thing that I _don't_ like about putting them in examples is that it implies that they aren't viable.  The peakrider strategy in particular could be sufficient for many users who don't want to write much/any code.  The fact that it's called "moneybot" kind of implies (like the other bots we have used) it should run and do something cool with minimal changes.

This leaves us with something we want to both run as a CLI app that "just works" and also a library that is easy to use to write strategies.  I don't have a specific answer here yet but I don't think it would be ideal to move the project further from "clone and run".

**Comment by harperj** [comment]
FWIW I think this is a more core question about the project direction so it's probably not really a blocker on this PR

**Comment by elsehow** [comment]
> The thing that I don't like about putting them in examples is that it implies that they aren't viable. The peakrider strategy in particular could be sufficient for many users who don't want to write much/any code. The fact that it's called "moneybot" kind of implies (like the other bots we have used) it should run and do something cool with minimal changes.

This is a good point. We should open up a separate issue about how to get up and running with the "clone-and-run" workflow.

File: moneybot/fund.py [hunk 4]
```diff
@@ -9,8 +9,8 @@
 
 import pandas as pd
 
-from moneybot.markets.adapters import MarketAdapter
-from moneybot.markets.history import MarketHistory
+from moneybot.market.adapters import MarketAdapter
+from moneybot.market.history import MarketHistory
```
Comment (elsehow) [inline moneybot/fund.py]: :+1:

**Comment by adamrothman** [comment]
@elsehow @harperj Maybe we could do something like what `cryptography` does [here](https://cryptography.io/en/latest/hazmat/primitives/) – a warning that says something like:

>The strategies included herein are viable for live-trading use, but crypto trading is serious business. It's entirely possible to lose a lot of money, very quickly. The authors of this library provide these for the sake of illustration and assume no responsibility for their performance if deployed.

**Pull request merged**